R/pkg/R/types.R

```diff
@@ -0,0 +1,41 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# types.R. This file handles the data type mapping between Spark and R
+
+# The primitive data types, where names(PRIMITIVE_TYPES) are Scala types whereas
+# values are equivalent R types.
+PRIMITIVE_TYPES <- c(
+  "byte"="integer",
+  "tinyint"="integer",
+  "integer"="integer",
+  "float"="numeric",
+  "double"="numeric",
+  "numeric"="numeric",
```

Comment:
I'm concern with this. decimal does not map exactly to numeric
And it says it is not supported:
http://spark.apache.org/docs/latest/sql-programming-guide.html#data-types